base/reflection.jl

```diff
@@ -1624,9 +1624,12 @@ function code_typed_by_type(@nospecialize(tt::Type);
         throw(ArgumentError("'debuginfo' must be either :source or :none"))
     end
     tt = to_tuple_type(tt)
-    matches = _methods_by_ftype(tt, #=lim=#-1, world)::Vector
     asts = []
-    for match in matches
+    matches = Core.Compiler.findall(tt, Core.Compiler.method_table(interp))
+    if matches === nothing
+        return asts
```

Comment:
It should not be an empty list

Reply:
Maybe we should raise an error for this case?